backend/src/StamAcasa.Api/Controllers/ProfileController.cs

```diff
@@ -165,5 +166,21 @@ public async Task<IActionResult> UpdateFamilyProfile(int id, [FromBody] UserProf
 
             return new OkObjectResult(result != null);
         }
+
+        /// <summary>
+        /// Delete user profile and related data.
+        /// </summary>
+        /// <returns></returns>
+        [HttpDelete]
+        public async Task<IActionResult> DeleteProfile()
```

Comment:
this endpoint should be called only by IdentityServer . If someone with valid token will call this endpoint he will delete from just from User table and will not delete from AspNetUsers

Reply:
in order to implement this we could add a claim named `admin` or `idsrv` and add a policy for this endpoint

Reply:
Very good points, Thank you!